test_dataframe_cow_chain - TypeError: Invalid value '100' for dtype string

https://github.com/coiled/benchmarks/actions/runs/6253354999/job/16994905070?pr=999

```
___________________________ test_dataframe_cow_chain ___________________________
[gw3] linux -- Python 3.9.18 /usr/share/miniconda3/envs/test/bin/python

value = 100, pa_type = DataType(string), dtype = string[pyarrow]

    def convert_fill_value(value, pa_type, dtype):
        if value is None:
            return value
        if isinstance(value, (pa.Scalar, pa.Array, pa.ChunkedArray)):
            return value
        if is_array_like(value):
            pa_box = pa.array
        else:
            pa_box = pa.scalar
        try:
>           value = pa_box(value, type=pa_type, from_pandas=True)

/usr/share/miniconda3/envs/test/lib/python3.9/site-packages/pandas/core/arrays/arrow/array.py:812: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pyarrow/scalar.pxi:1120: in pyarrow.lib.scalar
    ???
pyarrow/error.pxi:144: in pyarrow.lib.pyarrow_internal_check_status
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   pyarrow.lib.ArrowTypeError: Expected bytes, got a 'int' object

pyarrow/error.pxi:123: ArrowTypeError

The above exception was the direct cause of the following exception:

small_client = <Client: 'tls://10.0.0.124:8786' processes=10 threads=20, memory=71.61 GiB>

    def test_dataframe_cow_chain(small_client):
        memory = cluster_memory(small_client)  # 76.66 GiB
    
        df = timeseries_of_size(
            memory // 2,
            start="2020-01-01",
            freq="600ms",
            partition_freq="12h",
            dtypes={
                **{i: float for i in range(40)},
                **{i: int for i in range(41, 80)},
                **{i: object for i in range(81, 120)},
            },
        )
        print_dataframe_info(df)
    
        result = (
>           df.rename(columns={1: 1000})
            .replace("x", "xxx")
            .fillna(100)
            .astype({50: "float"})
            .loc[:, slice(2, 100)]
        )

tests/benchmarks/test_dataframe.py:89: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/share/miniconda3/envs/test/lib/python3.9/site-packages/dask/dataframe/core.py:1726: in fillna
    meta = self._meta_nonempty.fillna(
/usr/share/miniconda3/envs/test/lib/python3.9/site-packages/pandas/core/frame.py:5[493](https://github.com/coiled/benchmarks/actions/runs/6253354999/job/16994905070?pr=999#step:8:494): in fillna
    return super().fillna(
/usr/share/miniconda3/envs/test/lib/python3.9/site-packages/pandas/core/generic.py:6983: in fillna
    new_data = self._mgr.fillna(
/usr/share/miniconda3/envs/test/lib/python3.9/site-packages/pandas/core/internals/managers.py:433: in fillna
    return self.apply(
/usr/share/miniconda3/envs/test/lib/python3.9/site-packages/pandas/core/internals/managers.py:352: in apply
    applied = getattr(b, f)(**kwargs)
/usr/share/miniconda3/envs/test/lib/python3.9/site-packages/pandas/core/internals/blocks.py:1868: in fillna
    new_values = self.values.fillna(value=value, method=None, limit=limit)
/usr/share/miniconda3/envs/test/lib/python3.9/site-packages/pandas/core/arrays/arrow/array.py:818: in fillna
    fill_value = convert_fill_value(value, self._data.type, self.dtype)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

value = 100, pa_type = DataType(string), dtype = string[pyarrow]

    def convert_fill_value(value, pa_type, dtype):
        if value is None:
            return value
        if isinstance(value, (pa.Scalar, pa.Array, pa.ChunkedArray)):
            return value
        if is_array_like(value):
            pa_box = pa.array
        else:
            pa_box = pa.scalar
        try:
            value = pa_box(value, type=pa_type, from_pandas=True)
        except pa.ArrowTypeError as err:
            msg = f"Invalid value '{str(value)}' for dtype {dtype}"
>           raise TypeError(msg) from err
E           TypeError: Invalid value '100' for dtype string

/usr/share/miniconda3/envs/test/lib/python3.9/site-packages/pandas/core/arrays/arrow/array.py:815: TypeError
```

This most likely was caused by #992 
CC @phofl

--- a/tests/benchmarks/test_dataframe.py
+++ b/tests/benchmarks/test_dataframe.py
@@ -88,7 +88,7 @@ def test_dataframe_cow_chain(small_client):
     result = (
         df.rename(columns={1: 1000})
         .replace("x", "xxx")
-        .fillna(100)
+        .fillna({i: 100 for i in range(10, 70)})
         .astype({50: "float"})
         .loc[:, slice(2, 100)]
     )

